src/Smalot/PdfParser/Page.php

```diff
@@ -216,8 +216,59 @@ public function getText(self $page = null): string
         return '';
     }
 
+    /**
+     * Return true if the current page is a (setasign\Fpdi\Fpdi) FPDI/FPDF document
+     *
+     * @return bool true is the current page is a FPDI/FPDF document
+     */
+    public function isFpdf(): bool
+    {
+        if (\array_key_exists('Producer', $this->document->getDetails()) &&
+            \is_string($this->document->getDetails()['Producer']) &&
```

Comment:
When is `Producer` not a string?

Reply:
Well, I don't remmember if that could be null.  But I may just taking care that str_starts_with is apply to a String.  Do you want me to take it out?

Reply:
> Do you want me to take it out?

I would like to discuss implications for this if-clause first. We should document background information (maybe with a link to the spec?) so that in the future developer know what it means. For instance, are there other information we can use to determine whether something is produced by FPdf or not? I know that is additional work, anything in this direction is appreciated!

Reply:
Understood.  I use two sources of the information:
### 1. The Pdf 1.7 Specification.   [PDF 32000-1:2008](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf)
**Chapter 14: Document Interchange, Section 14.3: Metada, subSection: 14.3.3 Document Information Dictionary** 
**Page: 550**
It defines: 
> **Key:** Producer **Type:** text string **Value:** (Optional) If the document was converted to PDF from another format, the name of the conforming product that converted it to PDF.

It also has an example of values in the same page.

### 2. The FPDF source code. The file **fpdf.php**
> **FPDF Version:** 1.83 **Date:** 20021-04-18 **Author:** Olivier Plathey

In the line: 1834 it defines the function _putinfo():

```
protected function _putinfo()
{
	$this->metadata['Producer'] = 'FPDF '.FPDF_VERSION;
	$this->metadata['CreationDate'] = 'D:'.@date('YmdHis');
	foreach($this->metadata as $key=>$value)
		$this->_put('/'.$key.' '.$this->_textstring($value));
}
```
Here the metadata with key: **'Producer'** it set to **"FPDF 1.83"**
and in the line 1875 it defines the function **_enddoc()**, in this functions it set all the properties and values of the document. This include the line 1883: **`$this->_putinfo();`**

So with this 2 information, we know that the metadata with the key = 'Producer' will have the text value: 'FPDF ' . FPDF_VERSION, for any document generated/saved by FPDF.

Reply:
Let me know if with this is Ok.